feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -261,59 +240,66 @@ func TestHardwarecards(t *testing.T) {
 			rrValidation:          false,
 			operStatus:            "",
 			parent:                "",
-			pType:                 componentType["transceiver"],
+			pType:                 transceiverType.String(),
 		},
 	}}
 
 	for _, tc := range cases {
 		t.Run(tc.desc, func(t *testing.T) {
-			r, err := regexp.Compile(tc.regexpPattern)
-			if err != nil {
-				t.Fatalf("Cannot compile regular expression: %v", err)
-			}
-			cards := findMatchedComponents(t, dut, r)
+			cards := components.FindComponentsByType(t, dut, tc.componentType)
 			t.Logf("Found card list for %v: %v", tc.desc, cards)
 
 			if len(cards) == 0 {
 				t.Fatalf("Get card list for %q) on %v: got 0, want > 0", tc.desc, dut.Model())
 			}
-			ValidateComponentState(t, dut, cards, tc.cardFields)
+			validateComponentState(t, dut, cards, tc.cardFields)
+
 		})
 	}
 }
 
 func TestSwitchChip(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
-
-	regexpPattern := "^SwitchChip"
-	cardFields := properties{
-		descriptionValidation: false,
-		idValidation:          true,
-		nameValidation:        true,
-		partNoValidation:      true,
-		serialNoValidation:    false,
-		mfgNameValidation:     false,
-		mfgDateValidation:     false,
-		swVerValidation:       false,
-		hwVerValidation:       false,
-		fwVerValidation:       true,
-		operStatus:            "",
-		parent:                "",
-		pType:                 componentType["switchchip"],
-	}
-
-	r, err := regexp.Compile(regexpPattern)
-	if err != nil {
-		t.Fatalf("Cannot compile regular expression: %v", err)
+	var cardFields properties
+	switch dut.Vendor() {
+	case ondatra.JUNIPER:
```

Comment:
tests should not be vendor-specific without a documented deviation

Reply:
Done. Added new deviations